frontend/functions/ui_sections.py

```
@@ -321,15 +313,40 @@ def render_progress_page(
 
+    # Collect a unique list of activity types
+    unique_types = grouped_df['type'].unique()
+
+    # Handle colour mapping logic
+    # Case 1: Only one type
+    if len(unique_types) == 1:
+        color_discrete_sequence = ["#fc4c02"]
+
+    # Case 2: Multiple types — keep "Run" orange, let Plotly handle the rest
+    else:
+        # Get Plotly’s default color palette
```

Ruff (0.14.2)

Comment contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF003)